lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -21,23 +22,27 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
-  def self.upgrade_message
-    Puppet.warning <<-UPGRADE
-The current Puppet version is outdated and uses a library that was
-previously necessary on the current Ruby verison to support a colored console.
-
-Unfortunately this library prevents the DSC module from using a shared
-PowerShell process to dramatically improve the performance of resource
-application.
-
-To enable these improvements, it is suggested to upgrade to any x64 version of
-Puppet (including 3.x), or to a Puppet version newer than 3.x.
-    UPGRADE
-  end
-
+  POWERSHELL_UPGRADE_MSG = <<-UPGRADE
```

Comment:
This will be done in the MODULES-4014 ticket as it likely needs a test and QA review, as we are restricting on version where we didn't before